language_service/src/state.rs

```diff
@@ -186,12 +200,16 @@ impl<'a> CompilationStateUpdater<'a> {
     async fn load_manifest(
         &self,
         doc_uri: &Arc<str>,
-    ) -> Option<(Arc<str>, Vec<(Arc<str>, Arc<str>)>)> {
+    ) -> Option<(Arc<str>, Vec<(Arc<str>, Arc<str>)>, LanguageFeatures)> {
```

Comment:
nitpick: this type is starting to get long (which I'm often guilty of myself), so it seems like it might be worth putting it in a struct to make the intentions clear and use named access (rather than the `.0`, `.1`, `.2` below). Mostly a personal preference, but when I see `.0` or `.1` I fine with it because its just a pair, but once it gets to `.2` it starts looking messy to me.